episodes/hpc.Rmd

```diff
@@ -0,0 +1,295 @@
+---
+title: 'Deploying Targets on HPC'
+teaching: 10
+exercises: 2
+---
+
+```{R, echo=FALSE}
+# Exit sensibly when Slurm isn't installed 
+if (!nzchar(Sys.which("sbatch"))){
+  knitr::knit_exit("sbatch was not detected. Likely Slurm is not installed. Exiting.")
+}
+```
+
+:::::::::::::::::::::::::::::::::::::: questions 
+
+- Why would we use HPC to run Targets workflows?
+- How can we run Targets workflows on Slurm?
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::: objectives
+
+- Be able to generate a report using `targets`
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::: instructor
+
+Episode summary: Show how to write reports with Quarto
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+```{r}
+#| label: setup
+#| echo: FALSE
+#| message: FALSE
+#| warning: FALSE
+library(targets)
+library(tarchetypes)
+library(quarto) # don't actually need to load, but put here so renv catches it
+source("https://raw.githubusercontent.com/joelnitta/targets-workshop/main/episodes/files/functions.R?token=$(date%20+%s)") # nolint
+
+# Increase width for printing tibbles
+options(width = 140)
+```
+
+## Advantages of HPC
+
+If your analysis involves computationally intensive or long-running tasks such as training machine learning models or processing very large amounts of data, it will quickly become infeasible to use a single machine to run this.
+If you are part of an organisation with access to a High Performance Computing (HPC) cluster, you can easily leverage the numerous machines with Targets to scale up your analysis.
+This differs from the exucution we have learned so far, which spawns extra R processes on the *same machine* to speed up execution.
+
+## Configuring Targets for Slurm
+
+Fortunately, using HPC is as simple as changing the Targets `controller`.
+In this section we will assume that our HPC uses Slurm as its job scheduler, but you can easily use other schedulers such as PBS/TORQUE, Sun Grid Engine (SGE) or LSF.
```

Comment:
```suggestion
In this section we will assume that our HPC uses Slurm as its job scheduler, but you can use other schedulers such as PBS/TORQUE, Sun Grid Engine (SGE) or LSF.
```
Could you also elaborate on how this would be done?